range.go

```diff
@@ -217,6 +243,144 @@ func splitComparatorVersion(s string) (string, string, error) {
 	return strings.TrimSpace(s[0:i]), s[i:], nil
 }
 
+// getWildcardType will return the type of wildcard that the
+// passed version contains
+func getWildcardType(vStr string) wildcardType {
+	parts := strings.Split(vStr, ".")
```

Comment:
That can't happen as `getWildcardType` is executed on each splitted version, meaning that, with your example, `getWildcardType` will be called with `1.0.0` and `1.x.x`.